packages/neuron-wallet/src/startup/sync-block-task/task.ts

```diff
@@ -1,86 +1,45 @@
 import { remote } from 'electron'
 import { initConnection as initAddressConnection } from 'database/address/ormconfig'
-import AddressService from 'services/addresses'
-import LockUtils from 'models/lock-utils'
 import AddressesUsedSubject from 'models/subjects/addresses-used-subject'
-import BlockListener from 'services/sync/block-listener'
 import { NetworkWithID } from 'services/networks'
 import { register as registerTxStatusListener } from 'listeners/tx-status'
 import { register as registerAddressListener } from 'listeners/address'
+import IndexerRPC from 'services/indexer/indexer-rpc'
+import Utils from 'services/sync/utils'
 
-import { initDatabase } from './init-database'
+import { switchNetwork as syncSwitchNetwork } from './sync'
+import { switchNetwork as indexerSwitchNetwork } from './indexer'
 
 // register to listen address updates
 registerAddressListener()
 
-const {
-  nodeService,
-  addressDbChangedSubject,
-  addressesUsedSubject,
-  databaseInitSubject,
-  walletCreatedSubject,
-} = remote.require('./startup/sync-block-task/params')
+const { addressesUsedSubject, databaseInitSubject } = remote.require('./startup/sync-block-task/params')
 
 // pass to task a main process subject
 AddressesUsedSubject.setSubject(addressesUsedSubject)
 
-// maybe should call this every time when new address generated
-// load all addresses and convert to lockHashes
-export const loadAddressesAndConvert = async (): Promise<string[]> => {
-  const addresses: string[] = (await AddressService.allAddresses()).map(addr => addr.address)
-  const lockHashes: string[] = await LockUtils.addressesToAllLockHashes(addresses)
-  return lockHashes
-}
-
-// call this after network switched
-let blockListener: BlockListener | undefined
-export const switchNetwork = async () => {
-  // stop all blocks service
-  if (blockListener) {
-    await blockListener.stopAndWait()
+export const testIndexer = async (url: string): Promise<boolean> => {
```

Comment:
It'd be more clearer to name it such as `isIndexerEnabled` or something like that.

Reply:
ok